dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/response/SolrSearchLocationResult.java

```diff
@@ -49,14 +54,27 @@ public Map<String, String> getCustomData() {
     // -------------------- PRIVATE --------------------
 
     // Calculate center from the first two polygon points (representing a rectangle)
-    private GeoPoint calculateCenter(GeoPoint pointA, GeoPoint pointB) {
-        if (pointA == null || pointB == null) {
-            throw new IllegalArgumentException("To determine a center we need two points.");
-        }
+    private GeoPoint calculateCenter(List<GeoPoint> geoPoints) {
```

Comment:
The method keeps getting complex. Maybe we can add a test that will check the calculated center?
```
// given
var coords = ...
// when
var locationResult = new SolrSearchLocationResult(coords);
// then
assertThat(locationResult.getMarker().getLatitude()).isEqualTo(...);
assertThat(locationResult.getMarker().getLongitude()).isEqualTo(...);
```
with cases with one, two and more than two `GeoPoint`s